homeassistant/components/systemmonitor/sensor.py

```diff
@@ -500,8 +490,8 @@ async def async_setup_entry(
                 )
 
     scan_interval = DEFAULT_SCAN_INTERVAL
-    await async_setup_sensor_registry_updates(hass, sensor_registry, scan_interval)
     async_add_entities(entities)
+    await async_setup_sensor_registry_updates(hass, sensor_registry, scan_interval)
```

Comment:
Maybe don't add them until the first update is successful to avoid having to rely on restore entity?

Reply:
Something isn't working as expected with Restore entity anyhow so I need to figure out some way of dealing with this. Would prefer to have the data already before adding the entities so maybe should read entity registry directly instead.